### PR TITLE
added '<Card.Subheader>' component

### PR DESCRIPTION
**Pull request by dorduch**
### 🔦 Summary
Added a new `<Card.Subheader>` component

### ✅ Checklist
- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

File: src/Card/Subheader/Subheader.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    return (
+      <div className={styles.container}>
+        <Box verticalAlign="middle" flexGrow={1} overflow="hidden">
+          {isString(title) ? (
+            <Text ellipsis weight="normal" size="medium" dataHook="title">
+              {title}
+            </Text>
+          ) : (
+            <span data-hook="title-node">{title}</span>
```
[the remark this answers is not in the document]
Comment (dorduch) [reply src/Card/Subheader/Subheader.js]: nope. implemented this like `<Header>` is implemented. Will remove it

File: src/Card/Subheader/Subheader.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    title: oneOfType([node, string]).isRequired,
+    suffix: node,
+  };
+
+  render() {
+    const { title, suffix } = this.props;
+
+    return (
+      <div className={styles.container}>
+        <Box verticalAlign="middle" flexGrow={1} overflow="hidden">
```
[the remark this answers is not in the document]
Comment (dorduch) [reply src/Card/Subheader/Subheader.js]: for the text ellipsis. if there was no overflow then the left part will 'push' the right part out of the card

**Comment by dorduch** [comment]
1. I think `Subheader` (for example https://material-ui.com/api/list-subheader/)

**Comment by dorduch** [comment]
2. @shlomitc  added docs

File: src/Card/Subheader/Subheader.uni.driver.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { baseUniDriverFactory } from 'wix-ui-test-utils/base-driver';
+
+export const cardSubheaderDriverFactory = base => {
+  return {
+    ...baseUniDriverFactory(base),
+    /** Get the title text */
+    title: async () => base.$('[data-hook="title"]').text(),
+    /** Get the title node */
+    titleNode: async () => base.$('[data-hook="title-node"]'),
```
Comment (shlomitc) [inline src/Card/Subheader/Subheader.uni.driver.js]: please add a test for this

File: src/Card/Subheader/Subheader.uni.driver.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    /** Get the title node */
+    titleNodeByDataHook: async dataHook => base.$(`[data-hook=${dataHook}]`),
```
Comment (shlomitc) [inline src/Card/Subheader/Subheader.uni.driver.js]: internal `data-hook`s should not be exposed to our consumers, so I wouldn't encourage to expose such functions.
`getTitleNode` and `getSuffixNode` should be enough.

File: src/Card/Subheader/docs/storySettings.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { Category } from '../../../../stories/storiesHierarchy';
+
+export const storySettings = {
+  category: Category.COMPONENTS,
+  storyName: 'Card Subheader',
```
Comment (shlomitc) [inline src/Card/Subheader/docs/storySettings.js]: I would call it `Card.Subheader`, to match the component name.
if it was a standalone component to be imported, I would call it `CardSubheader`

File: src/Card/Subheader/docs/index.story.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  storyName: storySettings.storyName,
+  component: Subheader,
+  componentPath: '../Subheader.js',
+
+  componentProps: {
+    title: 'Card Subheader',
+  },
+
+  sections: [
+    tab({
```
Comment (shlomitc) [inline src/Card/Subheader/docs/index.story.js]: add
```js
importExample({
  source: `
import Card from 'wix-style-react/Card';
const Subheader = Card.Subheader;
`,
}),

**Pull request merged**